.NET C# no way to enable different .NET sdk's on .netcore projects! (#2289)
premake/premake-core: website/docs/dotnetsdk.md

Selects a .NET Sdk

```lua
dotnetsdk "sdk"
```

[overview](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/overview)

## parameters ##
`sdk` is one of
 * [web](https://learn.microsoft.com/en-us/aspnet/core/razor-pages/web-sdk?toc=%2Fdotnet%2Fnavigate%2Ftools-diagnostics%2Ftoc.json&bc=%2Fdotnet%2Fbreadcrumb%2Ftoc.json)
 * [razor](https://learn.microsoft.com/en-us/aspnet/core/razor-pages/sdk?toc=%2Fdotnet%2Fnavigate%2Ftools-diagnostics%2Ftoc.json&bc=%2Fdotnet%2Fbreadcrumb%2Ftoc.json)
 * [worker](https://learn.microsoft.com/en-us/dotnet/core/extensions/workers)
 * [blazor](https://learn.microsoft.com/en-us/aspnet/core/blazor/)
 * [windowsdesktop](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props-desktop?view=aspnetcore-8.0)
 * [mstest](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-mstest-sdk)

## mstest ##

> to make the MSTest sdk work you need to add the version to `global.json`:

Reviewer: Would it make sense to generate that file too?

Author: yes i was thinking about that but how would i make it so you could specify the version? or do i just have a default version and tell people where they can change it! and code for writing the file as i am not familliar with the file api's of premake

> to make the MSTest sdk work you need to add the version to `global.json`:

Author: yes that is correct but specifiying it becomes a little hard so i decided on the global.json route

Selects a .NET SDK

```lua
dotnetsdk "SDK"
```

For more information see the MSDN documentation [here](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/overview)

> ## parameters ##

Reviewer 2: ```suggestion
## Parameters ##
```

`SDK` is one of:

 * [Default](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props)
 * [Web](https://learn.microsoft.com/en-us/aspnet/core/razor-pages/web-sdk?toc=%2Fdotnet%2Fnavigate%2Ftools-diagnostics%2Ftoc.json&bc=%2Fdotnet%2Fbreadcrumb%2Ftoc.json)
 * [Razor](https://learn.microsoft.com/en-us/aspnet/core/razor-pages/sdk?toc=%2Fdotnet%2Fnavigate%2Ftools-diagnostics%2Ftoc.json&bc=%2Fdotnet%2Fbreadcrumb%2Ftoc.json)
 * [Worker](https://learn.microsoft.com/en-us/dotnet/core/extensions/workers)
 * [Blazor](https://learn.microsoft.com/en-us/aspnet/core/blazor/)
 * [WindowsDesktop](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props-desktop?view=aspnetcore-8.0)

>  * [MSTest](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-mstest-sdk)
> 
> 
> ## mstest ##
> when using MSTest sdk a version has to be specified!

Reviewer 2: ```suggestion
 * [MSTest](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-mstest-sdk):  Requires a version be specified.
```

### Applies To ###

Project configurations.

### Availability ###

Premake 5.0 beta4 or later.

Visual studio is the only toolset currently supported.

### Examples ###
```lua
dotnetsdk "Web"
```

```lua
dotnetsdk "Web/3.4.0"

> ```

Reviewer 2: These two lines should be swapped around - the empty line should be outside the snippet.

> ## CustomSDK
> 
> ```lua
> premake.api.addAllowed("dotnetsdk", "CustomSDK") -- add the customSDK to allowed values for dotnetsdk
> dotnetsdk "CustomSDK/3.4.0"
> ```
> 
> ```lua
> premake.api.addAllowed("dotnetsdk", "CustomSDK") -- add the customSDK to allowed values for dotnetsdk
> dotnetsdk "CustomSDK"
> ```

Reviewer 2: The extra title is a bit weird and would be better served as just a comment like this:
````suggestion
A custom SDK can be specified using the following:

```lua
premake.api.addAllowed("dotnetsdk", "CustomSDK") -- add the custom SDK to allowed values for dotnetsdk
dotnetsdk "CustomSDK"

-- Specifying a version with a custom SDK is also supported
dotnetsdk "CustomSDK/3.4.0"
```
````
